elfpy/pricing_models/base.py

```diff
@@ -344,13 +345,13 @@ def check_input_assertions(
             "pricing_models.check_input_assertions: ERROR: "
             f"expected quantity.amount >= {WEI}, not {quantity.amount}!"
         )
-        assert market_state.share_reserves >= 0, (
```

Comment:
seems like the outcome of just testing more cases. the one we breached was the normalized time variable. I added it to `share_reserves` and `bond_reserves` because it felt like a missing equivalence to what we were already doing elsewhere. but looking at it now, it's redundant, so I'm just going to remove these two here, in favor of the check in `update_market`: `self.market_state.check_market_non_zero()`

Reply:
fixed here 81a5a0d402c3d721322c93a94d08ce6d89b8acbd